### PR TITLE
etcd: upgrade go version from 1.21.4

### PR DESCRIPTION
**Pull request by arjunmalhotra1**
To keep etcd projects up to date with the latest patch releases & incorporate the latest security updates.

### CODE REVIEW

**Comment by serathius** [comment]
Please sign DCO

**Comment by arjunmalhotra1** [comment]
Thanks @serathius. Fixed it.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/bbolt
 
-go 1.21
+go 1.21.4
```
Comment (jmhbnz) [inline go.mod:3]: Hey @arjunmalhotra1 - I don't think it is agreed to start pinning `go.mod` to a specific patch version. The community has only just started discussing it in https://github.com/etcd-io/etcd/issues/16828.

Please keep the `go.mod` at `1.21` for now.

**Review by jmhbnz — CHANGES_REQUESTED** [review]
Hey @arjunmalhotra1 - Thanks for raising this, can you also please update:

```bash
cmd/bbolt/README.md:  Go Version: go1.21.1
.github/workflows/tests_windows.yml:          go-version: "1.21.1"
.github/workflows/tests_windows.yml:          go-version: "1.21.1"
```

**Comment by ahrtr** [comment]
Please rebase this PR.

FYI. we just merged https://github.com/etcd-io/bbolt/pull/608. Now we only need to update the go version in one place 
https://github.com/etcd-io/bbolt/blob/master/.go-version#L1

**Comment by arjunmalhotra1** [comment]
Thank you @ahrtr I have addressed the issue.

**Review by jmhbnz — APPROVED** [review]
LGTM on green - Thanks @arjunmalhotra1 

Nit - Ideally we would also update `cmd/bbolt/README.md:  Go Version: go1.21.1` but it's not a big deal.

**Comment by arjunmalhotra1** [comment]
Thanks @jmhbnz. I have updated in the readme as well.

**Review by fuweid — APPROVED** [review]
LGTM

**Review by serathius — APPROVED** [review]

**Pull request merged**